Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/Surveys.kt

```diff
@@ -34,18 +38,39 @@ class Surveys @Inject constructor(
     }
 
     suspend fun requestDetails(type: Type): Survey {
+        val surveyConfig = appConfigProvider.getAppConfig().survey
+        Timber.v("Requested survey: %s", surveyConfig)
+        /* no check here:
+         * if surveyOnHighRisk is not enabled, this use case shouldn't have been started in the first place
+         */
+//        if (type == Type.HIGH_RISK_ENCOUNTER && !surveyConfig.surveyOnHighRiskEnabled) {
+//            throw SurveyException(SurveyException.Type.HIGH_RISK_NOT_ENABLED)
+//        }
+        otpRepo.otpAuthorizationResult?.apply {
+            if (authorized && redeemedAt.toDateTime().monthOfYear() == Instant.now().toDateTime().monthOfYear()) {
+                throw SurveyException(SurveyException.Type.ALREADY_PARTICIPATED_THIS_MONTH)
+            }
+        }
+        val otp = otpRepo.otp ?: otpRepo.generateOTP()
+        val errorCode = surveyServer.authOTP(
+            otp,
+            deviceAttestation.attest(object : DeviceAttestation.Request {
+                override val scenarioPayload: ByteArray
+                    get() = otp.payloadForRequest
+            })
```

Comment:
Could you please simplify this : 
```kotlin
            deviceAttestation.attest {
                otp.payloadForRequest
            }
```

by doing:
```kotlin
    fun interface Request {
        /**
         * e.g. for EventSurvey, a UUID, base64 encoded.
         */
        fun scenarioPayload(): ByteArray
    }
```

in tests:
```kotlin
    data class TestAttestationRequest(
        private val scenarioPayload: ByteArray
    ) : DeviceAttestation.Request {

        override fun scenarioPayload(): ByteArray = scenarioPayload
.....
}
```

Reply:
thank you! changed it!